src/Middleware/ThrottleMiddleware.php

```diff
@@ -195,7 +197,31 @@ protected function _touch(): int
             );
         }
 
-        return Cache::increment($this->_identifier, 1, static::$cacheConfig) ?: 0;
+        return Cache::increment($this->_identifier, $this->_getRequestWeight($request), static::$cacheConfig) ?: 0;
+    }
+
+    /**
+     * Returns configured weight of each request, result of callable, or 1 when option
+     * is misconfigured or the callable did not return integer >= 0
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request ServerRequestInterface instance
+     * @return int
+     */
+    protected function _getRequestWeight(ServerRequestInterface $request): int
+    {
+        $configWeight = $this->getConfig('requestWeight');
+
+        if (is_callable($configWeight)) {
+            $configWeight = $configWeight($request);
+        }
+
+        if (!is_int($configWeight) || (is_int($configWeight) && $configWeight < 0)) {
```

Comment:
The `is_int($configWeight)` check is redundant. If it's not an integer the `!is_int($configWeight)` would have already succeeded and the expression after `||` won't be evaluated.